SayTheirNames/Source/Controller/Home/HomeController.swift

```diff
@@ -24,15 +24,32 @@
 
 import UIKit
 
-// MARK: - IDENTIFIERS
-private let headerIdentifier = "PersonHeaderCell"
-private let peopleIdentifier = "PersonCell"
+extension PersonsResponsePage: PaginatorResponsePage {
+    typealias Element = Person
+}
 
 final class HomeController: UIViewController {
     @DependencyInject private var network: NetworkRequestor
     
     required init() {
+        
+        weak var wself: HomeController?
+        
+        self.paginator =
+            Paginator<Person, PersonsResponsePage>(pageLoader: { (link: Link?, completion: @escaping (Result<PersonsResponsePage, Error>) -> Void) in
+            guard let sself = wself else { return }
+            
```

Comment:
Can we do something about the `wself` and `sself`?

Reply:
`self.paginator =
            Paginator<Person, PersonsResponsePage>(pageLoader: { [weak self](link, @escaping (Result<PersonsResponsePage, Error>) -> Void) in
                guard let self = self else {return}` i meant something like this

Reply:
You’re right, my bad 😄
Amazing work BTW 🚀

Reply:
@hybridcattt would you mind considering using weakSelf and get rid of wself/sself? Ie something like this

```
        weak var weakSelf: HomeController?
        
        self.paginator =
            Paginator<Person, PersonsResponsePage>(pageLoader: { (link: Link?, completion: @escaping (Result<PersonsResponsePage, Error>) -> Void) in
            guard let self = weakSelf else { return }
            
            // call network
            if let link = link {
                self.network.fetchPeopleWithLink(link, completion: completion)
            }
            else {
                self.network.fetchPeople(completion: completion)
            }
        })
        super.init(nibName: nil, bundle: nil)
        
        weakSelf = self
```

Reply:
sure that makes sense, done 👌